### ISSUE TITLE
  Error when trying python caimanmanager.py install

### ISSUE THREAD

**Issue by baharrrr** [flatironinstitute/CaImAn#217]
After installing caiman, I tried "python caimanmanager.py install" but I get the error that. 
ModuleNotFoundError: No module named 'caiman.source_extraction.cnmf.oasis'
File is located in the correct location, and I am not quiet sure what is the issue!

I appreciate your help.

**Comment by epnev** [comment]
Can you type 
```
echo $PYTHONPATH
```
and let us know what it gives? If you've used caiman in the past and have set PYTHONPATH you'll need to remove that for the new installer

**Comment by baharrrr** [comment]
This is what I get: /PATH TO/anaconda3/lib/python3.6

**Comment by baharrrr** [comment]
Now this is the error I get, env: python\r: No such file or directory

**Comment by pgunn** [comment]
What OS are you on? Can you cut'n'paste the command you typed as well as the full response?

**Comment by baharrrr** [comment]
$cd CaImAn
$ source activate caiman
(caiman)$ caimanmanager.py install
env: python\r: No such file or directory
 
I am on a mac OS 10.12.6

**Comment by pgunn** [comment]
Two things strike me as odd about that:
1) python should have been installed by conda, and should have been added to your path when you did the `source activate caiman`. It seems something went wrong with this
2) You should not cd into the caiman directory before activating caiman

The first is more of a problem; can you send me the output of `echo $PATH` ?

**Comment by baharrrr** [comment]
-bash: /Users/user/anaconda3/envs/caiman/bin:/opt/local/bin:/opt/local/sbin:/Users/baharrahsepar/anaconda3/bin:/Users/user/anaconda3/bin:/Users/user/anaconda3/bin:/Users/baharrahsepar/anaconda/bin:/anaconda/bin:/Users/user/anaconda/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/Users/user/Library/Enthought/Canopy_64bit/User/bin:/Library/Frameworks/Python.framework/Versions/3.4/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/TeX/texbin: No such file or directory

**Comment by pgunn** [comment]
That's the output from typing
echo $PATH
? It shouldn't be telling you "No such file or directory" from that.

**Comment by baharrrr** [comment]
/Users/user/anaconda3/envs/caiman/bin:/opt/local/bin:/opt/local/sbin:/Users/user/anaconda3/bin:/Users/user/anaconda3/bin:/Users/baharrahsepar/anaconda3/bin:/Users/user/anaconda/bin:/anaconda/bin:/Users/user/anaconda/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/Users/user/Library/Enthought/Canopy_64bit/User/bin:/Library/Frameworks/Python.framework/Versions/3.4/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/TeX/texbin

**Comment by baharrrr** [comment]
Sorry my mistake it was from the line above that I forgot to type echo

**Comment by pgunn** [comment]
Ok, and if you type

which python

on the commandline, what do you get?

**Comment by baharrrr** [comment]
/Users/user/anaconda3/envs/caiman/bin/python

**Comment by pgunn** [comment]
If you type
/usr/bin/env python
from the commandline, do you get a `>>>` prompt? If you do, type exit() to get back to your shell, and cut and paste the response here

**Comment by baharrrr** [comment]
Yes, 
Python 3.6.5 | packaged by conda-forge | (default, Apr  6 2018, 13:44:09) 
[GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53)] on darwin
Type "help", "copyright", "credits" or "license" for more information.

**Comment by pgunn** [comment]
Ok. That's good to see.

So, if you can close that terminal window, open a new one, and try the following commands (do not change directories or do other things first):

source activate caiman
caimanmanager.py install

and paste the response, that'd be great.

**Comment by baharrrr** [comment]
I get the same response as before 
env: python\r: No such file or directory

**Comment by pgunn** [comment]
Can you send me what you see if you do the following command?

head -5 `which caimanmanager.py`

**Comment by baharrrr** [comment]
I get:
head: caimanmanager.py: No such file or directory

**Comment by pgunn** [comment]
Sorry, I updated the command since I typed it initially. Include the backtics.

**Comment by baharrrr** [comment]
/Users/user/anaconda3/envs/caiman/bin/caimanmanager.py

**Comment by pgunn** [comment]
head -5 /Users/user/anaconda3/envs/caiman/bin/caimanmanager.py

**Comment by baharrrr** [comment]
#!/usr/bin/env python

import argparse
import distutils.dir_util
import filecmp

**Comment by pgunn** [comment]
What happens if you do `python /Users/user/anaconda3/envs/caiman/bin/caimanmanager.py install`

**Comment by baharrrr** [comment]
Now it did the work! Thank you! Could you tell me what was the issue?

**Comment by pgunn** [comment]
I'm a little confused about that myself. Your path indicates that you have installed conda several times into different places, the current install is into a fairly weird place, and the python you're getting from scripts is not the same one you're getting from the commandline. If you get the chance to clean that up (it might not be super easy to do that - if you have someone locally that might help you with cleaning up that and your dotfiles, you might want to ask them about it).

The other thing you might want to take a quick look at is your ~/.gitconfig (if you have one); there may be a autocrlf setting in there that's set to something unusual.

**Comment by baharrrr** [comment]
I see, thanks a lot for that.
 I will look into them.

**Comment by MilesQLi** [comment]
I met the same issue. When the current path is not CaImAn repository, caimanmanager.py cannot be found and when you are in the CaImAn repository, the caimanmanager.py used for this command is not the one: /Users/user/anaconda3/envs/caiman/bin/caimanmanager.py. So the full path has to be given.

**Comment by pgunn** [comment]
Activating an environment in Conda should handle adding its bin directory to your path. If that's not happening for you, we should look into why ; can you send the output of "echo $PATH" after activating the environment?

**Comment by MilesQLi** [comment]
I just did it. The directory is indeed in my path. So if I use "caimanmanager.py install" there would be no problem. But "python caimanmanager.py install" would cause that problem.

**Comment by MilesQLi** [comment]
The reason baharrr failed for "caimanmanager.py install" with the reason "env: python\r: No such file or directory" is because  EOLs are different for different OSs.

**Comment by pgunn** [comment]
Hm. I wonder if this happens with the commandline git tools, or only with a graphical git tool; there's a setting in git (that I thought defaulted to translating to local crlf) that controls that. I haven't ever seen this with my testing on OSX, but I use the commandline git.

https://help.github.com/articles/dealing-with-line-endings/
Miles, are you seeing this with your install? If so, can you tell me what git you're using and if you're using the commandline git, if you can paste your gitconfig, that'd be helpful.

**Comment by MilesQLi** [comment]
I use git on windows which is downloaded here: https://git-scm.com/download/win. 

In fact, I can run "caimanmanager.py install" on a windows console after activate caiman instead of a git (mingw64) console.

The git config is as follows:
<details>
core.symlinks=false
core.autocrlf=true
core.fscache=true
color.diff=auto
color.status=auto
color.branch=auto
color.interactive=true
help.format=html
rebase.autosquash=true
http.sslcainfo=C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt
http.sslbackend=openssl
diff.astextplain.textconv=astextplain
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
credential.helper=manager
credential.helper=manager
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true

**Comment by pgunn** [comment]
Looks like you have autocrlf set to true, which is the right thing; we'll need to see if baharrr has a different setting.

**Comment by pgunn** [comment]
@baharrrr Hi, if you use gitter, if you can grab my attention on the channel, we can dig into what happened with your git checkout.

**Comment by pgunn** [comment]
@baharrrr Hi, any chance you can join us on the gitter chat room? If you can't, let me know, but I think these issues are caused by your having done the checkout with an unusual CRLF setting for your git client; I can help you debug that, but it's not actually an issue with Caiman so much as with your local settings.

**Comment by johannesmaxkappel** [comment]
I had the same error as baharrr, in that I could not execute `python caimanmanager.py install` but instead, as @pgunn suggested,  `python /Users/user/anaconda3/envs/caiman/bin/caimanmanager.py install`  did the trick. So this thread was very helpful! 

I did **not** have the issue though that I got the error message env: python\r: No such file or directory but I had the same initial error 'ModuleNotFoundError: No module named 'caiman.source_extraction.cnmf.oasis'' . Running caiman on Windows 10.0.15063 64bit

**Comment by johannesmaxkappel** [comment]
Unfortunately, when I run the test with caimanmanager.py, I still get 'ModuleNotFoundError: No module named 'caiman.source_extraction.cnmf.oasis''

What could be the reason for that? I definitely have the cython file at the correct location.

**Comment by pgunn** [comment]
Hi Johannes,
In the conda environment, the pip install of caiman should install caimanmanager.py in your path. You shouldn't invoke it with python first, just invoke it by its name (when you call python on something, it doesn't search your path to find it). Don't do this in the caiman sourcedir, do it someplace else.

**Comment by johannesmaxkappel** [comment]
OK I don't fully understand your reply. I ran, as the install instructions say, first `pip install .`. And it did install a file caimanmanger.py in my CaImAn path. Then, as suggested in the install instructions, I ran `python caimanmanager.py install`.  

I got this Error, same as baharrrr: ModuleNotFoundError: No module named 'caiman.source_extraction.cnmf.oasis'

Instead, I ran `python /Users/user/anaconda3/envs/caiman/bin/caimanmanager.py install`, which worked. Then I tried to run `python caimanmanager.py test`. That gave me the same error again with the oasis. Funny enough, I am currently using that caiman install and it works like a charm...
